pkg/manifests/manifests_test.go

```diff
@@ -779,26 +780,45 @@ func TestPrometheusOperatorConfiguration(t *testing.T) {
 		t.Fatalf("expected node selector to be master, got %q", got)
 	}
 
-	expectedPromOpImage := "docker.io/openshift/origin-prometheus-operator:latest"
-	resPromOpImage := d.Spec.Template.Spec.Containers[0].Image
-	if resPromOpImage != expectedPromOpImage {
-		t.Fatalf("Configuring the Prometheus Operator image failed, expected: %v, got %v", expectedPromOpImage, resPromOpImage)
-	}
-
 	prometheusReloaderFound := false
 	prometheusWebTLSCipherSuitesArg := ""
 	prometheusWebTLSVersionArg := ""
-	for i := range d.Spec.Template.Spec.Containers[0].Args {
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") {
-			prometheusReloaderFound = true
-		}
+	kubeRbacProxyTLSCipherSuitesArg := ""
+	kubeRbacProxyMinTLSVersionArg := ""
+	for i, container := range d.Spec.Template.Spec.Containers {
+		if container.Name == "prometheus-operator" {
+			if d.Spec.Template.Spec.Containers[i].Image != "docker.io/openshift/origin-prometheus-operator:latest" {
+				t.Fatalf("%s image incorrectly configured", container.Name)
+			}
+			for j := range d.Spec.Template.Spec.Containers[i].Args {
+				if strings.HasPrefix(d.Spec.Template.Spec.Containers[i].Args[j], PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") {
+					prometheusReloaderFound = true
+				}
+
+				if strings.HasPrefix(d.Spec.Template.Spec.Containers[i].Args[j], PrometheusOperatorWebTLSCipherSuitesFlag) {
+					prometheusWebTLSCipherSuitesArg = d.Spec.Template.Spec.Containers[i].Args[j]
+				}
 
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSCipherSuitesFlag) {
-			prometheusWebTLSCipherSuitesArg = d.Spec.Template.Spec.Containers[0].Args[i]
+				if strings.HasPrefix(d.Spec.Template.Spec.Containers[i].Args[j], PrometheusOperatorWebTLSMinTLSVersionFlag) {
+					prometheusWebTLSVersionArg = d.Spec.Template.Spec.Containers[i].Args[j]
+				}
+			}
 		}
 
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSMinTLSVersionFlag) {
-			prometheusWebTLSVersionArg = d.Spec.Template.Spec.Containers[0].Args[i]
+		if container.Name == "kube-rbac-proxy" {
+			if d.Spec.Template.Spec.Containers[i].Image != "docker.io/openshift/origin-kube-rbac-proxy:latest" {
+				t.Fatalf("%s image incorrectly configured", container.Name)
+			}
+
+			for j := range d.Spec.Template.Spec.Containers[i].Args {
```

Comment:
Should we extract a function `containsArg` or something similar for improved readability?

Reply:
+1

Reply:
Not sure if we need function. Looks like this could help with readability. WDYT?

```go
containsArgs :=  d.Spec.Template.Spec.Containers[i].Args
for j := range containsArgs {
```

Reply:
This still leaves the deeply nested for + if statements. Even a double nested for loop is a code-smell, and here we have 4-5 levels of nesting. Extracting the for loop into a single function will reduce the nesting depth. 

In addition, we are repeating this code chunk too often which is error prone, especially because it is part of the tests.

Reply:
@slashpai in case it is useful, we already have a similar extraction in the e2e tests - https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/config_test.go#L708

Reply:
Ya that makes sense. I will update

Reply:
addressed